zebra-consensus/src/block/tests.rs

```diff
@@ -16,14 +16,13 @@ use zebra_chain::{
 };
 use zebra_test::transcript::{TransError, Transcript};
 
-static VALID_BLOCK_TRANSCRIPT: Lazy<Vec<(Arc<Block>, Result<block::Hash, TransError>)>> =
+static INVALID_GENESIS_BLOCK_TRANSCRIPT: Lazy<Vec<(Arc<Block>, Result<block::Hash, TransError>)>> =
     Lazy::new(|| {
         let block: Arc<_> =
             Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..])
                 .unwrap()
                 .into();
-        let hash = Ok(block.as_ref().into());
-        vec![(block, hash)]
+        vec![(block, Err(TransError::Any))]
     });
```

Comment:
This looks like a valid genesis block, can you add a comment explaining why it doesn't work?

Reply:
Also, if we're not testing any success cases, then these tests aren't as useful as they were before.

Do we need a finalized block transcript runner?

Reply:
it doesn't work because genesis blocks are finalized and this is attempting to treat it as non finalized state by inserting it with `CommitBlock`. I'll have to double check how its currently failing but eventually I think we'd want to make this an explicit error as in "Error, you inserted a block which we know must be finalized"

Reply:
As an alternative, we could just pass the genesis block directly from CommitBlock to CommitFinalizedBlock, as a special case.

Here are the specific changes I'd like to see to this test:
- [ ] make any required fixes to the state implementation
- [ ] add a test that succeeds
- [ ] if this test still fails after the state fixes, add a comment to the test transcript, explaining why the test fails (because it looks like it should succeed)